Katana/Store.swift

```diff
@@ -62,24 +96,46 @@ private func emptyStateInitializer<S: State>() -> S {
 }
 
 /**
- The `Store` class contains and manages the state of the application and the
- updates to the state itself.
+ The `Store` is a sort of single entry point that handles the logic of your application.
+ In Katana, all the various pieces of information that your application manages should be store
+ in a single atom, called state (see also `State` protocol).
  
- In Katana, every relevant information in the application should be stored in the
- Store's state. The only way to update the state is to dispatch an `Action`. At this point
- the store will execute the following operations:
+ The `Store`, however, doesn't really implements any application specific logic: this class
+ only manages operations that are requsted by the application-specific logic. In particular,
+ you can require the `Store` to execute something by `dispatching a dispatchable item`.
+
+ Currently the stores handles 3 types of dispatchable: `State Updater`, `Side Effect` and (deprecated) `Action`.
  
- - it executes the middleware
- - it creates the new state, by invoking the `updateState` function of the action
+ #### Update the state
+ As written before, in Katana every relevant information in the application should be stored in the
+ Store's state. The only way to update the state is to dispatch a `StateUpdater` (or the legacy `Action`).
+ At this point the `Store` will execute the following operations:
+ 
+ - it executes the interceptors
```

Comment:
I think that it creates the new state before invoking the interceptors... otherwise how can you pass `prevState` and `currentState` to the `onStateChange`interceptors?
The second-last step is then set the newly created state to the `state` variable and finally it resolves the promise.

Reply:
Doc is correct.

In the interceptors you can do
```
state = getState()
next()
newState = getState()
```